src/client.js

```diff
@@ -0,0 +1,188 @@
+import rest from './rest.js';
+
+const PODMAN_SYSTEM_ADDRESS = "/run/podman/podman.sock";
+
+export function getAddress(system) {
+    if (system)
+        return PODMAN_SYSTEM_ADDRESS;
+    const xrd = sessionStorage.getItem('XDG_RUNTIME_DIR');
+    if (xrd)
+        return (xrd + "/podman/podman.sock");
+    console.warn("$XDG_RUNTIME_DIR is not present. Cannot use user service.");
```

Comment:
Looks like this message will be shown over and over again. Could it  lazily initialize and remember the session address to avoid that?

Reply:
not really, because we try `/info` endpoint for root and rootless at first and only if that succeeds, then we connect to it.
If `$XDG_RUNTIME_DIR` is not present, then we cannot connect to rootless and thus no other call will be done for rootless. (You cannot download image for it, we won't load any container/images and won't listen for events).